### DISCUSSION TITLE
Type of `Union` is more descriptive `T | String` than `T | (string & {})`

### DISCUSSION THREAD

**Discussion by TomokiMiyauci** [chakra-ui/chakra-ui#5031] [Ideas] [answered]
I learned the hacky way `T | (string & {})` through this library. Thanks.

I found a way that works the same way, `T | String`. Isn't this one more descriptive?

**Comment by anubra266** [comment]
@TomokiMiyauci could you give more context, like where you found this code?

**Reply by TomokiMiyauci** [reply]
@anubra266 Oops sorry.

https://github.com/chakra-ui/chakra-ui/blob/baa6fee280530137548f961db5cb308100bc3920/packages/styled-system/src/utils/types.ts#L14

From the user's point of view, `(string & {})` looks a little strange.

**Reply by TKDev7** [reply]
Out of curiosity, what does string & {} mean?

**Reply by TomokiMiyauci** [reply]
@MrHBS (string & {}) prevents upcasting of string literal types.

As you can see, the following example will upcast to type `string` as of TypeScript 4.4.2.

```ts
type Color = 'red' | 'blue' | string // string
````

(string & {}) can accept a `string`, but apply the string literal intellisense.

```ts
type Color = 'red' | 'blue' | (string & {}) 
// IntelliSense 'red' | 'blue', but accept any string
```

**Comment by anubra266** [comment, marked as answer]
You're right @TomokiMiyauci

**Reply by TomokiMiyauci** [reply]
@anubra266 Thanks!
